### PR TITLE
Missing mails

### PR DESCRIPTION
**Pull request by noirbizarre**
This PR:
- add mail testing facilities
- fix the mailer choice (use FakeMailer when debugging or testing and real mailer in production)
- test issues and discussions signals
- test issues and discussions mails
- fix discussions mails not being sent.

I think another PR is required to test all the mails and, even better, group test by modules.

### CODE REVIEW

File: udata/core/discussions/models.py [hunk 1]
```diff
@@ -46,5 +46,5 @@ def person_involved(self, person):
     @property
     def external_url(self):
         return self.subject.url_for(
-            _anchor='discussion-{ id }'.format(id=self.id),
+            _anchor='discussion-{id}'.format(id=self.id),
```
Comment (davidbgk) [inline udata/core/discussions/models.py]: So the actual bug was here? :(

File: udata/mail.py [hunk 2]
```diff
@@ -14,11 +16,21 @@
 
 mail = Mail()
 
+mail_sent = signal('mail-sent')
+
+
+class FakeMailer(object):
```
Comment (davidbgk) [inline udata/mail.py]: :+1:

**Comment by davidbgk** [comment]
r+

File: udata/core/discussions/models.py (hunk 1 above)
Comment (noirbizarre) [inline udata/core/discussions/models.py]: Yes, the spaces were used in the key resolution :( (we are too used too Jinja and Vue)

**Pull request merged**